exec/java-exec/src/test/java/org/apache/drill/TestUnionAll.java

```diff
@@ -1197,4 +1197,64 @@ public void testFieldWithDots() throws Exception {
       .baselineValues("1", "2", "1", null, "a")
       .go();
   }
-}
\ No newline at end of file
+
+  @Test
+  public void testUnionAllRightEmptyDir() throws Exception {
```

Comment:
1. Does union (aka union distinct) also work fine?
2. Case when bot directories in union / union all are empty?